### PR TITLE
kubeadm: Idempotent service account creation.

### PR DESCRIPTION
**Pull request by pipejakob**
**What this PR does / why we need it**:
During `kubeadm init`, ignore errors for duplicates when creating service accounts.

https://github.com/kubernetes/kubeadm/issues/278

Fixes: https://github.com/kubernetes/kubeadm/issues/288

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/apiconfig/clusterroles_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package apiconfig
```
Comment (luxas) [inline cmd/kubeadm/app/phases/apiconfig/clusterroles_test.go]: license needed

File: cmd/kubeadm/app/phases/apiconfig/clusterroles.go [hunk 2]
```diff
@@ -64,7 +65,9 @@ func CreateServiceAccounts(clientset *clientset.Clientset) error {
 
 	for _, sa := range serviceAccounts {
 		if _, err := clientset.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Create(&sa); err != nil {
-			return err
+			if !apierrors.IsAlreadyExists(err) {
+				return err
+			}
```
Comment (luxas) [inline cmd/kubeadm/app/phases/apiconfig/clusterroles.go]: if already exists, call `.Update()`

Comment (pipejakob) [inline cmd/kubeadm/app/phases/apiconfig/clusterroles.go]: Will calling Update() actually do anything for these service accounts? Based on the creation code above, they just have a name and a namespace, so I thought an Update would just be a no-op. If the service accounts exist, they should already have the same name and namespace, no?

**Comment by pipejakob** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

File: cmd/kubeadm/app/phases/apiconfig/clusterroles.go (hunk 2 above)
Comment (luxas) [inline cmd/kubeadm/app/phases/apiconfig/clusterroles.go]: Of course!

Yeah, that's correct. I did it mechanically locally without thinking of this special case... this SGTM

**Comment by luxas** [comment]
@dchen1107 This is fixing a part of https://github.com/kubernetes/kubeadm/issues/288 and is clearly a bugfix. Required for upgrading kubeadm from v1.6 to v1.7

I'm adding the v1.7 label

**Comment by luxas** [comment]
@pipejakob Needs a rebase

**Comment by pipejakob** [comment]
Rebased!

**Comment by luxas** [comment]
/lgtm
/approve no-issue

(no-issue due to https://github.com/kubernetes/test-infra/issues/2975)

**Pull request merged**